app/legacy/dao/notifications_dao.py

```diff
@@ -22,11 +36,137 @@ async def get_notification(id: UUID4) -> Row[Any]:
         Args:
             id (UUID4): id of the notification
 
+        Raises:
+            NonRetryableError: Unable to get the notification
+
         Returns:
-            Row: notification table row
+            Row[Any]: notification table row
         """
         legacy_notifications = metadata_legacy.tables['notifications']
         stmt = select(legacy_notifications).where(legacy_notifications.c.id == id)
 
-        async with get_read_session_with_context() as session:
-            return (await session.execute(stmt)).one()
+        @db_retry
+        async def _get() -> Row[Any]:
+            async with get_read_session_with_context() as session:
+                return (await session.execute(stmt)).one()
```

Comment:
I believe we need to catch exceptions here and raise retryable / nonretryable exceptions instead.

Reply:
Fixed, thank you!